lib/src/Uncertainty/Algorithm/Simulation/NAIS.cxx

```diff
@@ -0,0 +1,224 @@
+
+  Collection<Distribution> margins(dimensionSample);
+
+  // Computation of auxiliary distribution using ot.Mixture
+  for (UnsignedInteger k = 0; k < dimensionSample ; ++k)
+  {
+    Collection<Distribution> collectionOfDistribution(numberOfSample_);
+    for (UnsignedInteger i = 0; i < numberOfSample_ ; ++i)
+    {
+      collectionOfDistribution[i] = Normal(sample(i, k), silverman[k]);
+    }
+    margins[k] = Mixture(collectionOfDistribution, weights_);
+  } // for k
+  return ComposedDistribution(margins);
+}
+
+// Function computing weigths of sample
+Point NAIS::computeWeights(const Sample & sample,
+                           const Sample & outputSample,
+                           const Scalar eventThresholdLocal,
+                           const Distribution & auxiliaryDistribution)
+{
+  Indices indiceCritic(0);
+  for (UnsignedInteger i = 0; i < sample.getSize(); ++i)
+  {
+    const Bool weightBool = getEvent().getOperator()(outputSample(i, 0), eventThresholdLocal);
+    if (weightBool)
+      indiceCritic.add(i);
+  } // for i
+
+  // Extract the relevant sample
+  const Sample criticalSample(sample.select(indiceCritic));
+  // Compute initial distribution logPDF in parallel
+  const Sample initialLogPDF = initialDistribution_.computeLogPDF(criticalSample);
+  // Compute initial distribution logPDF in parallel
+  const Sample auxilliaryLogPDF = auxiliaryDistribution.computeLogPDF(criticalSample);
+  weights_ = Point(sample.getSize());
+  for (UnsignedInteger i = 0; i < criticalSample.getSize(); ++i)
+  {
+    weights_[indiceCritic[i]] = std::exp(initialLogPDF[i][0] - auxilliaryLogPDF[i][0]);
+  }
+  return weights_;
+}
+
+// Main function that computes the failure probability
+void NAIS::run()
+{
+
+  // Drawing of samples using initial density
+  sample_ = initialDistribution_.getSample(numberOfSample_);
+
+  // Evaluation on limit state function
+  Sample outputSample(getEvent().getFunction()(sample_));
+
+  // Computation of current quantile
+  Scalar currentQuantile = outputSample.computeQuantile(rhoQuantile_)[0];
+  Distribution auxiliaryDistribution;
+  if (getEvent().getOperator()(currentQuantile, getEvent().getThreshold()))
+  {
+    currentQuantile = getEvent().getThreshold();
+    auxiliaryDistribution = initialDistribution_;
+  }
+  else
+  {
+    // Computation of weights_
+    weights_ = computeWeights(sample_, outputSample, currentQuantile, initialDistribution_);
+
+    // Computation of auxiliary distribution
+    auxiliaryDistribution = computeAuxiliaryDistribution(sample_, weights_);
+  }
+
+  while ((getEvent().getOperator()(getEvent().getThreshold(), currentQuantile)) && (currentQuantile != getEvent().getThreshold()))
+  {
+    // Drawing of samples using auxiliary density
+    sample_ = auxiliaryDistribution.getSample(numberOfSample_);
+
+    // Evaluation on limit state function
+    outputSample = getEvent().getFunction()(sample_);
```

Comment:
Then, here you want to evaluate the function by `maximumOuterSampling` blocs of `blockSize` instead of the whole sample, something like:
```
sample_ = Sample(0, dim)
outputSample = Sample(0, 1)
for (UnsignedInteger i = 0; i < maximumOuterSampling; ++ i)
{
  Sample blockSample(auxiliaryDistribution.getSample(blockSize));
  sample_.add(blockSample);
  outputSample.add(getEvent().getFunction()(blockSample));
}
```
you can see this logic in SubsetSampling etc

Reply:
finally i i use maximumOuterSampling  and blockSize like this :

UnsignedInteger numberOfSample = getMaximumOuterSampling() * getBlockSize();
sample_ = auxiliaryDistribution.getSample(numberOfSample);

Reply:
my point was to evaluate the sample by blocs in order to control the number of simultaneous parallel evaluations of the function
it also allows to stop the simulation before finishing the whole N evaluations are over

Reply:
yes, but the evaluation by blocs is missing

Reply:
ok i understood. One more question : why the i is not used?
in 
const Sample blockSample(auxiliaryDistribution.getSample(blockSize));
blockSample have the same value in the loop for?

Reply:
it's only used to count the number of blocs added
we cannot preallocate the samples and do "sample[i] = blockSample" because it's a group of point, so we have to use add

Reply:
ok i try and the execution is different from above. the value of outputSample are differents with the new code. i push all the changes. the NAIS TEST failed

Reply:
that's normal, the samples depend of the size you ask